network/transport/v2/protocol.go

```diff
@@ -140,12 +140,22 @@ func (p *protocol) Configure(_ transport.PeerID) error {
 func (p *protocol) Start() (err error) {
 	p.ctx, p.cancel = context.WithCancel(context.Background())
 
-	// load old payload query jobs
-	if err = p.payloadScheduler.Run(); err != nil {
-		return fmt.Errorf("failed to start retrying TransactionPayloadQuery: %w", err)
+	nodeDID, err := p.nodeDIDResolver.Resolve()
+	if err != nil {
+		log.Logger().WithError(err).Errorf("failed to resolve node DID")
+	}
+
+	if nodeDID.Empty() {
+		log.Logger().Warn("not starting the payload scheduler as node DID is not set")
+	} else {
+		// load old payload query jobs
+		if err = p.payloadScheduler.Run(); err != nil {
```

Comment:
but this will still add the jobs to the scheduler? I thought we also discussed we shouldn't add them add all, to avoid writing lots of jobs that will never be ran/cleaned up when a node is just reading?

Reply:
Also: when using the "automatic node DID resolver" (default when not set and not in strict mode) the node DID might be available later when a DID document (with nutscomm endpoint) is created, without restarting. So technically more correct would be to start the scheduler, but not adding any jobs to it (when node DID is empty).

That would allow the node to start receiving private TXs after getting a node DID set, without restarting.

Might sound like an edge case, but I think it might happen in workshops/demos? (otherwise we'll have to make sure the manuals state you'll have to restart the node).

Reply:
> but this will still add the jobs to the scheduler? I thought we also discussed we shouldn't add them add all, to avoid writing lots of jobs that will never be ran/cleaned up when a node is just reading?

Oops. Indeed. Will fix

Reply:
> Might sound like an edge case, but I think it might happen in workshops/demos? (otherwise we'll have to make sure the manuals state you'll have to restart the node).

I'll add it to the manuals. This way the behaviour is at least deterministic.

Reply:
> Actually, this isn't great. Does it make sense to only do this when strict mode is enabled? You'd still get error messages on development. Not sure if that's an issue.

I also want my dev environments to be clean of errors.